File: node-sdk/multi_turn.mjs
/**
 * RoutePlex Node.js SDK — Multi-turn Conversation
 *
 * npm install @routeplex/node
 */

import { RoutePlex } from "@routeplex/node";

const client = new RoutePlex({ apiKey: process.env.ROUTEPLEX_API_KEY });

const response = await client.chat([
  { role: "system", content: "You are a helpful JavaScript tutor. Keep answers concise." },
  { role: "user", content: "What are arrow functions?" },
  { role: "assistant", content: "Arrow functions are a shorter syntax: const add = (a, b) => a + b;" },
  { role: "user", content: "How are they different from regular functions?" },
]);

console.log(response.output);

File: python-sdk/multi_turn.py
"""
RoutePlex Python SDK — Multi-turn Conversation

pip install routeplex
"""

import os
from routeplex import RoutePlex, Message

client = RoutePlex(api_key=os.environ["ROUTEPLEX_API_KEY"])

# Multi-turn with message history
response = client.chat([
    Message.system("You are a helpful Python tutor. Keep answers concise."),
    Message.user("What are list comprehensions?"),
    Message.assistant("List comprehensions are a concise way to create lists: [x**2 for x in range(10)]"),
    Message.user("Can I add conditions to them?"),
])

print(response.output)
